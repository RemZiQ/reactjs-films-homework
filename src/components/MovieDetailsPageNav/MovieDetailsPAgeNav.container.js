import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { getFilms, getGenres } from '../../modules/module_Search/actions';


import Nav from './MovieDetailsPageNav';

const mapStateToProps = store => ({ genres: store.genres });
const mapDispatchToProps = dispatch => ({
  fetchData: (search) => {
    dispatch(getFilms(search));
  },
  fetchGengres: () => {
    dispatch(getGenres());
  },
});

export default withRouter(connect(
  mapStateToProps, mapDispatchToProps,
)(Nav));
