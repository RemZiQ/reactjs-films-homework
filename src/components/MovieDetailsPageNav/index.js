export { default } from './MovieDetailsPAgeNav.container';
