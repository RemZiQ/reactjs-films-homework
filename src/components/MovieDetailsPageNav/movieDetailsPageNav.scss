.pageHeader__nav {
  display: flex;
  box-sizing: border-box;
  justify-content: space-between;
  width: 100%;
  min-height: 70px;
  max-height: 70px;
  padding: 0 100px 0 120px;
}

.logo {
  text-transform: uppercase;
  font-size: 50px;
  letter-spacing: 2px;
  line-height: 64px;
  color: #fff;
  font-family: "Helvetica Neue", sans-serif;
  font-weight: 800;
}

.nav__form_search {
  position: relative;
}

.nav__search {
  width: 520px;
  max-height: 70px;
  padding: 15px 0 20px 15px;
  background: none;
  border: solid 3px white;
  border-radius: 10px;
  font-size: 28px;
  letter-spacing: 1px;
  line-height: 44px;
  color: #fff;
  font-family: "Helvetica Neue", sans-serif;
  outline: none;
}
.nav__search::selection {
  background-color: rgba(10, 177, 233, 0.5);
  color: rgba(255, 255, 255, 0.5);
}

.nav__search_button {
  position: absolute;
  top: 5px;
  right: 15px;
  width: 200px;
  height: 60px;
  background: none;
  color: white;
  text-align: right;
  border: none;
}

.nav__search_button:focus {
  border: none;
  outline: none;
}

::-webkit-search-cancel-button {
  -webkit-appearance: none;
}

.nav__search_button::before {
  content: "\f002";
  font-family: "Font Awesome 5 Free";
  color: white;
  font-size: 30px;
  font-weight: bold;
}
