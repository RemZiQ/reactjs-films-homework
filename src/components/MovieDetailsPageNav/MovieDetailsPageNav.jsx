import React, { Component } from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { getFilms, getGenres } from '../../modules/module_Search/actions';

import './movieDetailsPageNav.scss';

class Nav extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchValue: '',
    };
  }

  fetchFilmsAndGenres = (e) => {
    const search = this.state.searchValue;
    this.setState({ searchValue: '' });
    const { fetchData, fetchGengres, genres } = this.props;
    // dont use preventDefault when use REACT ROUTE <LINK> !!!!!
    // e.preventDefault();
    if (!genres.length) {
      fetchGengres();
    }
    fetchData(search);
  };

  // submitForm = () => {
  //   this.setState
  // }

  handleChangeInput = (e) => {
    this.setState({
      searchValue: e.target.value,
    });
  }
  render() {
    return (
      <nav className="pageHeader__nav">
        <div className="logo">films</div>
        <form onSubmit={console.log('*****')} method="get" className="nav__form_search">
          <input 
            type="search" 
            className="nav__search" 
            id="searchInputID" 
            value={this.state.searchValue}
            onChange={this.handleChangeInput}
            />
          <Link to={{pathname:'/', search:`?query=${this.state.searchValue}`}}>
          {console.log("____", this.state.searchValue)}
          <button onClick={this.fetchFilmsAndGenres} type="submit" className="nav__search_button" />
          </Link>
        </form>
      </nav>
    );
  }
}

Nav.propTypes = {
  fetchData: propTypes.func.isRequired,
  fetchGengres: propTypes.func.isRequired,
  genres: propTypes.arrayOf(propTypes.object).isRequired,
};

const mapStateToProps = store => ({ genres: store.genres });
const mapDispatchToProps = dispatch => ({
  fetchData: (search) => {
    dispatch(getFilms(search));
  },
  fetchGengres: () => {
    dispatch(getGenres());
  },
});

export default connect(
  mapStateToProps, mapDispatchToProps,
)(Nav);
