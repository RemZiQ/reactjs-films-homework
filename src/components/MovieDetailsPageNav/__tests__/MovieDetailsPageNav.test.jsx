import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
// import { Provider } from 'react-redux';
// import { MovieDetailsPageFooter } from './MovieDetailsPageFooter';
import { Nav } from '../MovieDetailsPageNav';


it('Nav renders well', () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<Nav />);
  expect(result).toMatchSnapshot();
});

// it('renders well2222', () => {
//   const renderer = new ShallowRenderer();
//   const result = renderer.render(<MovieDetailsPageFooter store={{}} />);
//   expect(result).toMatchSnapshot();
// });

// it('renders well', () => {
//   const renderer = new ShallowRenderer();
//   const result = renderer.render(<MovieDetailsPageFooter store={null} />);
//   expect(result).toMatchSnapshot();
// });
