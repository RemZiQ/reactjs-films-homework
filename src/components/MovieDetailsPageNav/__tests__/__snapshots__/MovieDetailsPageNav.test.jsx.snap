// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders well 1`] = `
<nav
  className="pageHeader__nav"
>
  <div
    className="logo"
  >
    films
  </div>
  <form
    className="nav__form_search"
    method="get"
  >
    <input
      autoComplete="off"
      className="nav__search"
      id="searchInputID"
      onChange={[Function]}
      type="search"
      value=""
    />
    <Link
      to={
        Object {
          "pathname": "/search",
          "search": "?query=",
        }
      }
    >
      <button
        className="nav__search_button"
        onClick={[Function]}
        type="submit"
      />
    </Link>
  </form>
</nav>
`;
